src/datasets/table.py

```diff
@@ -24,10 +24,12 @@
 
 
 def inject_arrow_table_documentation(arrow_table_method):
-    def wrapper(method):
-        out = wraps(arrow_table_method)(method)
```

Comment:
Is it safe to replace `wraps` with manually setting `out.__doc__ = arrow_table_method.__doc__` ?
Otherwise, It was breaking the doc-builder's `inspect` usage.

Reply:
basically, are we using other properties set by `wraps` such as `__dict__`?

Reply:
@lhoestq @albertvillanova @mariosasko

Reply:
It's useful to have the same typing as well via `__annotations__ `, but `__dict__` is probably not important.

Feel free to also set the `__annotations__` as well here !

Reply:
handled in https://github.com/huggingface/datasets/pull/3690/commits/865261eaecbfbc18e265e16e9f62ce3f26713dff